src/main/java/geocode/kdtree/KDTree.java

```diff
@@ -44,7 +44,22 @@ public KDTree( List<T> items ) {
     }
 
     public T findNearest( T search ) {
-        return findNearest(root, search, 0).location;
+        return findNearest(search, null);
+    }
+
+    /**
+     * Finds nearest point within a max distance (in km)
+     */
+    public T findNearest(T search, Double maxDistance) {
+        T nearest = findNearest(root, search, 0).location;
+        if (null != maxDistance) {
+            int earthRadiusInKm = 6371;
```

Comment:
Constant values.... what if the size of the earth changes?!

Reply:
Note, software may become inconsistent after a million years

Reply:
😆 I'll pull this out to a string constant because I use it in my unit tests too.

Reply:
WGS-84 defines the radius as 6378.137km
https://en.wikipedia.org/wiki/World_Geodetic_System

Reply:
That's the radius at the equator. The 6371 value I'm using is the mean radius (https://en.wikipedia.org/wiki/Earth_radius). Do you think I should switch it?

Reply:
Mean radius is fine 👍